src/Parser/AggregateMoneyParser.php

```diff
@@ -1,51 +1,38 @@
 <?php
 
+declare(strict_types=1);
+
 namespace Money\Parser;
 
 use Money\Currency;
 use Money\Exception;
+use Money\Money;
 use Money\MoneyParser;
 
+use function sprintf;
+
 /**
  * Parses a string into a Money object using other parsers.
- *
- * @author Frederik Bosch <f.bosch@genkgo.nl>
  */
 final class AggregateMoneyParser implements MoneyParser
 {
     /**
      * @var MoneyParser[]
+     * @psalm-var non-empty-array<MoneyParser>
      */
-    private $parsers = [];
+    private array $parsers;
 
     /**
      * @param MoneyParser[] $parsers
+     * @psalm-param non-empty-array<MoneyParser> $parsers
      */
     public function __construct(array $parsers)
     {
-        if (empty($parsers)) {
-            throw new \InvalidArgumentException(sprintf('Initialize an empty %s is not possible', self::class));
-        }
-
-        foreach ($parsers as $parser) {
-            if (false === $parser instanceof MoneyParser) {
-                throw new \InvalidArgumentException('All parsers must implement '.MoneyParser::class);
-            }
-
-            $this->parsers[] = $parser;
-        }
+        $this->parsers = $parsers;
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function parse($money, $forceCurrency = null)
+    public function parse(string $money, Currency|null $forceCurrency = null): Money
     {
-        if ($forceCurrency !== null && !$forceCurrency instanceof Currency) {
```

Comment:
These notices have been removed from all over the library: much simpler to upgrade the type and let the type-checker tell that there's an incompatibility.